### ISSUE TITLE
Candidate Not Registered

### ISSUE THREAD

**Issue by grichardomi** [opencats/OpenCATS#466]
Using Firefox,

Login an existing candidate from the Career Page, I'm getting this message "You have not registered yet.  Please wait while we direct you to the job list"

Can anyone help with the cause of this message or how to register a candidate?

Thank you!

**Comment by RussH** [comment]
so I don't use the careers page, however;

https://demo.opencats.org/careers/

should reproduce what you want. If you haven't registered, you just browse the job listings, and then when you try to apply for one, you're prompted to  register.

**Comment by grichardomi** [comment]
Thanks for responding.

I still get the message after registering (admin@example.com | test4 | 84770) . And it is not allowing me to update my profile.

**Comment by shantadam** [comment]
Hi grichardomi,

Forgive me for going with your username, but I have no reference to your first name.

As for the issue, may I ask what is your role while accessing OpenCATS? Were you given access to a link? Are you a user? a candidate? An admin? If not an admin, do you have an admin? Are you trying to install? Is it an existing system?

We need some more in-depth knowledge of the context on who is accessing and how. Simply mentioning browser as Firefox, won’t help.

Regards,
Shant Adam,


On Feb 28, 2020, at 12:09 PM, grichardomi <notifications@github.com> wrote:

﻿

Using Firefox,

Login an existing candidate from the Career Page, I'm getting this message "You have not registered yet. Please wait while we direct you to the job list"

Can anyone help with the cause of this message or how to register a candidate?

Thank you!

**Comment by grichardomi** [comment]
what is your role while accessing OpenCATS? 
Candidate

Were you given access to a link?
https://demo.opencats.org/careers/
admin2@example.com | lastname: test5 | zipcode: 84770 

 Are you a user?
Candidate

 If not an admin, do you have an admin? 
Candidates should not have to have admin access in order to update their profile

Are you trying to install? 
I experience same behavior on demo site and on an newly installed server.

**Comment by shantadam** [comment]
I only have good news for you from my end. All my tests, through your provided link along with the candidate credentials passed successfully by logging in.

At this point, I have two favours to ask you:
- Please find the way to clear your specific browser’s cache.
- If issue persist, please provide your browser’s and OS detail specs. Not just names.

Regards,
Shant Adam


On Feb 29, 2020, at 5:03 PM, grichardomi <notifications@github.com> wrote:

﻿

what is your role while accessing OpenCATS?
Candidate

Were you given access to a link?
https://demo.opencats.org/careers/
admin2@example.com<mailto:admin2@example.com> | lastname: test5 | zipcode: 84770

Are you a user?
Candidate

If not an admin, do you have an admin?
Candidates should not have to have admin access in order to update their profile

Are you trying to install?
I experience same behavior on demo site and on an newly installed server.

**Comment by grichardomi** [comment]
No change with Browser cache cleared on Firefox.

As a candidate applying for job, I should be able to revise profile as needed.  When trying to  update, message is displayed:  "You have not registered yet.  Please wait while we direct you to the job list..."

**Comment by shantadam** [comment]
The fact is you just helped narrow the scenario down to a specific issue. Which is now much more clear.

At first your email read as if you were unable to access and upon login attempt you were getting the warning measage. However, it is now clear that the issue is beyong login access and it lies within the “candidate personal profile updating” action. The latter issue will have to be investigated, resolved and in terms of release I’ll relay the call to @RussH.

In the meantime, are you in a position to help chip-in with a patch on this issue? Let us know.

Regards,
Shant Adam


On Mar 1, 2020, at 9:42 AM, grichardomi <notifications@github.com> wrote:

﻿

No change with Browser cache cleared on Firefox.

As a candidate applying for job, I should be able to revise profile as needed. When trying to update, message is displayed: "You have not registered yet. Please wait while we direct you to the job list..."

**Comment by grichardomi** [comment]
I looking to enlist PHP consultants for a fix. So far, no positive feedback. Will keep any outcome here.

**Comment by grichardomi** [comment]
Will post any outcome here.

**Comment by grichardomi** [comment]
Had a consultant working on this. Here is the fix:

1)  *modules\careers\CareersUI.php*

2) Edit the line 1748 (This will change the preg pattern) to the below

3) /"([^\"]+)"="([^\"]*)"/

Correct condition is below
if (preg_match_all('/"([^\"]+)"="([^\"]*)"/', $_COOKIE[$id], $matches) > 0)

**Comment by shantadam** [comment]
Grichardomi,

It seems that you do have a team that backs you up.

Are you in a position to ask them to help contribute the changes to OpenCATS repo? If so, ask them to follow the following high level quick instructions, as such:

  *   Login Github.com with one of your developer’s account.
  *   Access https://github.com/opencats/OpenCATS/
  *   Fork off the project under their account.
  *   On their local development environment (computer)
  *   Git clone the repo locally in an independent folder
  *   Git checkout the “develop” branch
  *   Copy over and apply your described changes.
  *   Git commit with a clear message and some detail.
  *   Git push to the developer’s Github repo.
  *   Back to github.com
  *   Navigate to the developer’s opencast/OpenCATS page
  *   Click on “New Pull Request”
  *   Select base:develop and compare:develop
  *   Create Pull Request
  *   Add Reviewers (RussH and Shant Adam)

Regards,
Shant Adam

**Comment by grichardomi** [comment]
No, I'm a very small fish.

I hired a PHP consultant specifically for this fix.

**Comment by shantadam** [comment]
Fair enough, grichardomi. We’ll see to incorporate the fix.
Until the next release, thank you for your contribution.
Feel free to reach us anytime you have questions around OpenCATS.
Thank you.

Regards,
Shant Adam

**Comment by hansmach1ne** [comment]
While assessing application's security I stumbled upon the careers endpoint. It appears to me that application gives cookie after login on the /careers/index.php endpoint. The cookie format is the following (after URL decoding it): `cats1cw="email"="mach1ne@mach1ne.org""lastName"="mach1ne""zip"="10000"`,

Regex that is incorrectly validating the cookie is here: https://github.com/opencats/OpenCATS/blob/master/modules/careers/CareersUI.php#L262.

Said regex will match the cookie only when in this format: `\"email\"=\"mach1ne%40mach1ne.org\"\"lastName\"=\"mach1ne\"\"zip\"=\"10000\"`.

Bottom line is regex inside getCookieField function need to be modified (for the careers module at least), to `/"([^"]+)"="([^"]*)"/` as [grichardomi](https://github.com/grichardomi) provided will fix the issue.

**Comment by RussH** [comment]
@hansmach1ne @grichardomi  thank you for the commit - I'll test this in a local instance (& then assuming it's good I'll accept it)
